ansible/roles/mambaforge/tasks/mambaforge.yml

```diff
@@ -25,19 +17,37 @@
   failed_when: false
   register: conda_version_cmd
 
+- name: Wipe the existing installation if it differs from the target version
+  ansible.builtin.file:
+    path: "{{ mamba_home }}"
+    state: absent
+  when:
+    - conda_version_cmd['rc'] is defined
+    - conda_version_cmd['stdout'] is defined
+    - (conda_version_cmd['rc'] == 0) and (conda_version_cmd['stdout'] != 'conda ' ~ mambaforge_conda_version)
+
+- name: Verify that the installation directory doesn't exist
+  ansible.builtin.stat:
+    path: "{{ mamba_home }}"
+  register: mambaforge_home_check
+  failed_when: mambaforge_home_check['stat']['exists']
+
+- name: Create the installation directory
+  ansible.builtin.file:
+    path: "{{ mamba_home }}"
+    state: directory
+    owner: root
+    group: root
+    mode: "0755"
+
 - name: Execute the installation script
   ansible.builtin.command:
     argv:
       - /tmp/mambaforge.sh
       - -b
-      - -u
```

Comment:
I think the installer might actually fail if the directory already exists and no `-u` is passed.
(Meaning, the `-u` is said to be "update", but is actually "ignore the (non-) existence of the directory.)
So, we'd either retain the `-u` or not create the directory beforehand.

Reply:
Ah ok, didn't know that. In that case we can simplify things by leaving `-u` out and not bothering with my explicit check ("Verify that the installation directory doesn't exist").